src/evaluate.py

```diff
@@ -0,0 +1,128 @@
+parser.add_argument("--plot_interval", type=int, default=50, help="batch interval between saving generator sample visualisations")
+parser.add_argument("--n_downsample", type=int, default=2, help="number downsampling layers in encoder")
+parser.add_argument("--dim", type=int, default=32, help="number of filters in first encoder layer")
+
+opt = parser.parse_args()
+print(opt)
+
+cuda = True if torch.cuda.is_available() else False
+
+# Create inference output directories for both transfer directions
+os.makedirs("out_eval/%s/plot_A2B/" % opt.model_name, exist_ok=True)
+os.makedirs("out_eval/%s/plot_B2A/" % opt.model_name, exist_ok=True)
+
+input_shape = (opt.channels, opt.img_height, opt.img_width)
+
+# Dimensionality (channel-wise) of image embedding
+shared_dim = opt.dim * 2 ** opt.n_downsample
+
+# Initialize generator and discriminator
+encoder = Encoder(dim=opt.dim, in_channels=opt.channels, n_downsample=opt.n_downsample)
+shared_G = ResidualBlock(features=shared_dim)
+G1 = Generator(dim=opt.dim, out_channels=opt.channels, n_upsample=opt.n_downsample, shared_block=shared_G)
+G2 = Generator(dim=opt.dim, out_channels=opt.channels, n_upsample=opt.n_downsample, shared_block=shared_G)
+
+if cuda:
+    encoder = encoder.cuda()
+    G1 = G1.cuda()
+    G2 = G2.cuda()
+
+assert os.path.exists("saved_models/%s/encoder_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained encoder exists
+assert os.path.exists("saved_models/%s/G1_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained G1 exists
+assert os.path.exists("saved_models/%s/G2_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained G2 exists
+    
+# Load pretrained models
+encoder.load_state_dict(torch.load("saved_models/%s/encoder_%02d.pth" % (opt.model_name, opt.epoch)))
+G1.load_state_dict(torch.load("saved_models/%s/G1_%02d.pth" % (opt.model_name, opt.epoch)))
+G2.load_state_dict(torch.load("saved_models/%s/G2_%02d.pth" % (opt.model_name, opt.epoch)))
+
+# Set to eval mode 
+encoder.eval()
+G1.eval()
+G2.eval()
+
+Tensor = torch.cuda.FloatTensor if cuda else torch.Tensor
+dataloader = torch.utils.data.DataLoader(
+	DataProc(opt),
+	batch_size=opt.batch_size,
+	shuffle=True,
+	num_workers=opt.n_cpu,
+    pin_memory=True
+)
+
+feats = defaultdict(list)
+
+# ----------
+#  Testing
+# ----------
+
+progress = tqdm(enumerate(dataloader),desc='',total=len(dataloader))
+for i, batch in progress:
+
+    # Set model input
+    X1 = Variable(batch["A"].type(Tensor))
+    X2 = Variable(batch["B"].type(Tensor))
+
+    # -------------------------------
+    #  Infer with Encoder and Generators
+    # -------------------------------
+
+    # Get shared latent representation
+    mu1, Z1 = encoder(X1)
+    mu2, Z2 = encoder(X2)
+
+    # Reconstruct speech
+    recon_X1 = G1(Z1)
+    recon_X2 = G2(Z2)
+
+    # Translate speech
+    fake_X1 = G1(Z2)
+    fake_X2 = G2(Z1)
+
+    # Cycle translation
```

Comment:
Looks like you are not using cycle translation anywhere. Could be deleted as it is only used for training.

Reply:
That's a fair suggestion, but I'm using the cyclic reconstruction in the plot function. 

I think it might be good to keep just to have a full visualisation on what it learnt to do. Could help in observing performance changes with architectural modifications - for unseen data

Reply:
Agreed!